src/models/components/policy/tahh_normal_stochastic_policy.py

```diff
@@ -0,0 +1,22 @@
+import torch
+import torch.distributions as distributions
+import torch.nn as nn
+import torchrl
+from torch import Tensor
+from torch.distributions.distribution import Distribution
+
+from .stochastic_policy import StochasticPolicy
+
+
+class TanhNormalStochasticPolicy(StochasticPolicy):
+    def __init__(self, dim_input: int, dim_out: int):
+        super().__init__()
+        self.fc_mean = nn.Linear(dim_input, dim_out)
+        self.fc_std = nn.Linear(dim_input, dim_out)
+        self.softplus = nn.Softplus()
+
+    def forward(self, input: Tensor) -> Distribution:
+        mean = self.fc_mean(input)
+        std = self.softplus(self.fc_std(input)) + 1e-7  # std 0 causes error
+        tanh_norm_dist = torchrl.modules.TanhNormal(mean, std)
```

Comment:
元々の実装だとTanhNormalは-5 ~ 5の範囲なんですよね。なのでちょっと修正です。https://pytorch.org/rl/reference/generated/torchrl.modules.TanhNormal.html
```suggestion
        tanh_norm_dist = TanhNormal(mean, std, upscale=1.0)
```

Reply:
設定しました https://github.com/MLShukai/PrimitiveAMI/pull/16/commits/f23d77a0be29593299207a7bf30c8105c16e42cb